[ZING-JVM] SizeEstimatorTest.testExactHeapCostAfterUpdateWithMultipleBackupNodes & testSinglePut

**testExactHeapCostAfterUpdateWithMultipleBackupNodes**
```
java.lang.AssertionError: Heap cost calculation is wrong! expected:<156> but was:<188>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.map.SizeEstimatorTest.testExactHeapCostAfterUpdateWithMultipleBackupNodes(SizeEstimatorTest.java:69)
```
https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK6/34/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testExactHeapCostAfterUpdateWithMultipleBackupNodes/

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK7/55/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testExactHeapCostAfterUpdateWithMultipleBackupNodes/

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK8/56/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testExactHeapCostAfterUpdateWithMultipleBackupNodes/

**testSinglePut**
```
java.lang.AssertionError: expected:<156> but was:<188>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.SizeEstimatorTest.testSinglePut(SizeEstimatorTest.java:55)
```
https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK6/34/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testSinglePut/

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK7/55/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testSinglePut/

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK8/56/testReport/junit/com.hazelcast.map/SizeEstimatorTest/testSinglePut/

--- a/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HeapData.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HeapData.java
@@ -92,7 +92,6 @@ public class HeapData implements Data {
 
     @Override
     public int getHeapCost() {
-        // reference (assuming compressed oops)
         return REFERENCE_COST_IN_BYTES + (payload != null ? ARRAY_HEADER_SIZE_IN_BYTES + payload.length : 0);
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/map/SizeEstimatorTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/map/SizeEstimatorTest.java
@@ -30,6 +30,7 @@ import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import static com.hazelcast.util.JVMUtil.REFERENCE_COST_IN_BYTES;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
@@ -38,6 +39,11 @@ import static org.junit.Assert.assertTrue;
 public class SizeEstimatorTest extends HazelcastTestSupport {
 
     protected TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+    // the JVM-independent portion of the cost of Integer key + Long value record is 124 bytes
+    // (without taking into account 8 references to key, record and value objects)
+    private static final int JVM_INDEPENDENT_ENTRY_COST_IN_BYTES = 124;
+    // JVM-dependent total cost of entry
+    private static final int ENTRY_COST_IN_BYTES = JVM_INDEPENDENT_ENTRY_COST_IN_BYTES + 8 * REFERENCE_COST_IN_BYTES;
 
     @Test
     public void smoke() {
@@ -48,16 +54,14 @@ public class SizeEstimatorTest extends HazelcastTestSupport {
 
     @Test
     public void testSinglePut() {
-        long expectedPerEntryHeapCost = 156L;
         SizeEstimatorTestMapBuilder<Integer, Long> testMapBuilder = new SizeEstimatorTestMapBuilder<Integer, Long>(factory);
         IMap<Integer, Long> map = testMapBuilder.withNodeCount(1).withBackupCount(0).build(getConfig());
         map.put(0, 10L);
-        assertEquals(expectedPerEntryHeapCost, testMapBuilder.totalHeapCost());
+        assertEquals(ENTRY_COST_IN_BYTES, testMapBuilder.totalHeapCost());
     }
 
     @Test
     public void testExactHeapCostAfterUpdateWithMultipleBackupNodes() {
-        long expectedPerEntryHeapCost = 156L;
         int putCount = 1;
         int nodeCount = 1;
         SizeEstimatorTestMapBuilder<Integer, Long> testMapBuilder = new SizeEstimatorTestMapBuilder<Integer, Long>(factory);
@@ -66,7 +70,7 @@ public class SizeEstimatorTest extends HazelcastTestSupport {
             map.put(i, System.currentTimeMillis());
         }
         long heapCost = testMapBuilder.totalHeapCost();
-        assertEquals("Heap cost calculation is wrong!", expectedPerEntryHeapCost * putCount * nodeCount, heapCost);
+        assertEquals("Heap cost calculation is wrong!", ENTRY_COST_IN_BYTES * putCount * nodeCount, heapCost);
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/map/impl/recordstore/LazyEntryViewFromRecordTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/map/impl/recordstore/LazyEntryViewFromRecordTest.java
@@ -33,6 +33,7 @@ import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import static com.hazelcast.util.JVMUtil.REFERENCE_COST_IN_BYTES;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
@@ -41,6 +42,8 @@ import static org.mockito.Mockito.mock;
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
 public class LazyEntryViewFromRecordTest {
+
+    private static final int ENTRY_VIEW_COST_IN_BYTES = 97 + 3 * REFERENCE_COST_IN_BYTES;
 
     private final String key = "key";
     private final String value = "value";
@@ -80,7 +83,7 @@ public class LazyEntryViewFromRecordTest {
 
     @Test
     public void test_getCost() throws Exception {
-        assertEquals(109, view.getCost());
+        assertEquals(ENTRY_VIEW_COST_IN_BYTES, view.getCost());
     }
 
     @Test

